processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/MeasureFieldConverterImpl.java

```diff
@@ -57,11 +61,18 @@ public void convert(CarbonRow row, BadRecordLogHolder logHolder)
     String value = row.getString(index);
     Object output;
     boolean isNull = CarbonCommonConstants.MEMBER_DEFAULT_VAL.equals(value);
-    if (value == null || value.length() == 0 || isNull) {
+    if (value == null || isNull) {
       logHolder.setReason(
           "The value " + " \"" + value + "\"" + " with column name " + measure.getColName()
               + " and column data type " + dataType + " is not a valid " + dataType + " type.");
       row.update(null, index);
+    } else if (value.length() == 0) {
```

Comment:
both the condition could not be merged.
when value is empty there will be two decision to be take
1. empty is good 
2. empty data is bad